sql: fill pg_extension.geometry_columns/geography_columns

otan opened the pull request:
Release note (sql change): Populate the pg_extension.geometry_columns
and pg_extension.geography_columns virtual table with metadata for
tables which store have columns relevant to geometry or geography data
types.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49424)
<!-- Reviewable:end -->

rohany commented:
I'll take a closer look at this later, but the basic populate function is lazy -- you don't need a special generator

rohany commented on pkg/sql/pg_extension.go:
```diff
@@ -32,6 +36,64 @@ var pgExtension = virtualSchema{
+					if colDesc.Type.Family() != matchingFamily {
+						continue
+					}
+					m := colDesc.Type.InternalType.GeoMetadata
```
I think generally we don't expose access to InternalType

otan replied:
fixed.

rohany commented on pkg/sql/pg_extension.go:
```diff
@@ -32,6 +36,64 @@ var pgExtension = virtualSchema{
+						shapeName = geopb.Shape_Geometry.String()
+					}
+
+					if err := addRow(
```
does this table have oids?

otan replied:
nope.

rohany commented on pkg/sql/pg_extension.go:
```diff
@@ -32,6 +36,64 @@ var pgExtension = virtualSchema{
 	validWithNoDatabaseContext: false,
 }
 
+func postgisColumnsTablePopulator(
```
The paradigm we have is to define the populate function normally, and call out to a function that takes in `addRow` instead -- I think its a bit cleaner that way.

otan replied:
hmm, the problem then is i'd be duplicating the same code twice. seems cleaner to define this just once.

rohany replied:
Valid

rohany commented on pkg/sql/pg_extension.go:
```diff
@@ -44,9 +106,7 @@ CREATE TABLE pg_extension.geography_columns (
-	generator: func(ctx context.Context, p *planner, db *DatabaseDescriptor) (virtualTableGenerator, cleanupFunc, error) {
-		return nil, func() {}, errors.Newf("not yet implemented")
-	},
+	populate: postgisColumnsTablePopulator(types.GeographyFamily),
```
Are there any permissions that we need to consider here?

otan replied:
i added one fix

otan committed:
sql: fill pg_extension.geometry_columns/geography_columns

Release note (sql change): Populate the pg_extension.geometry_columns
and pg_extension.geography_columns virtual table with metadata for
tables which store have columns relevant to geometry or geography data
types.

otan commented:
bors r=rohany
